python/pyspark/mllib/linalg/distributed.py

```diff
@@ -297,6 +444,20 @@ def numCols(self):
         """
         return self._java_matrix_wrapper.call("numCols")
 
+    def computeGramianMatrix(self):
+        """
+        Computes the Gramian matrix `A^T A`. Note that this cannot be
+        computed on matrices with more than 65535 columns.
```

Comment:
We should maybe also add this note about max columns to the IndexedRowMatrix.scala for consistency.